shared/datastore.go

```diff
@@ -76,14 +76,16 @@ func LoadTestRunKeys(
 		}
 	}
 	var globalKeyFilter mapset.Set
-	if !IsLatest(sha) {
-		var ids TestRunIDs
-		if ids, err = loadKeysForRevision(ctx, baseQuery, sha); err != nil {
-			return nil, err
-		}
-		globalKeyFilter = mapset.NewSet()
-		for _, id := range ids {
-			globalKeyFilter.Add(id)
+	if len(revisions) > 1 || len(revisions) == 1 && !IsLatest(revisions[0]) {
+		for _, sha := range revisions {
+			var ids TestRunIDs
+			if ids, err = loadKeysForRevision(ctx, baseQuery, sha); err != nil {
+				return nil, err
+			}
+			globalKeyFilter = mapset.NewSet()
```

Comment:
Creating a new set in each iteration of the loop seems odd, if it's correct can you add a comment about why? Or not if I'm just being daft. If it is wrong, can a test be written for it?

Reply:
No, you're right - that's my bad. Left over from the original single-loop code. Added a unit test.